.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,71 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
```

Comment:
I think it would be beneficial to scan non-main branches prior to merging to main. Does GitHub advise against that?

Reply:
Ah looks like it runs on the _target_ branch for PRs 👍

Reply:
Yeah, CodeQL ran on this branch #inception 😂